tests/foreman/maintain/test_backup_restore.py

```diff
@@ -452,13 +452,14 @@ def test_positive_backup_restore(
 
     expected_files = get_exp_files(sat_maintain, skip_pulp)
     assert set(files).issuperset(expected_files), assert_msg
-
+    config_files = sat_maintain.execute(
+        f"tar -tf {backup_dir}/config_files.tar.gz"
+    ).stdout.splitlines()
     # Check if certificate tar file is present in Capsule backup.
     if instance == 'capsule':
```

Comment:
Should it also verify it isn't present otherwise? Or put in an other way: why is this `if` here?

I dislike assertions that are under `if` without an `else` because it's easy to make mistakes. For example, if you had a typo in `capsule` then the assertion is never executed.

Reply:
For this case, I guess assertion only applies to `capsule` test as only those have certs.tar but I understand your point. But I'm not sure if it's worth adding else part though.

Reply:
What's the harm in adding it? I've had too many masked failures that I think it's a good habit